src/internal_rep.rs

```diff
@@ -1766,6 +1770,27 @@ impl ValidatedCall {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct ValidatedModule {
+    pub name: CascadeString,
+    type_infos: Vec<TypeInfo>,
```

Comment:
This should be a Vec<&TypeInfo> rather than owning the TypeInfos.  That get rid of any potential problems if the TypeInfo is later changed, and eliminate the need for the clone() call at compile.rs:517